Objective-C/Tests/ReplicatorTest.m

```diff
@@ -2542,3 +2734,34 @@ - (void) testReplicatedDocument {
 }
 
 @end
+
+@implementation TestConflictResolver
+
+@synthesize winner=_winner;
+
+/**
+ this resolve will
+ (i) returns nil, if either local or remote is not present/deleted.
+ (ii) returns remote, if remote document contains key `pass` and it's value `true`
+ (iii) returns local, if local document contains key `pass` and it's value `true` && no remote
+     with same `pass` key.
+ (iv) returns nil, if neither  local nor remote contains the key with `pass` and value `true`.
+ */
```

Comment:
what do you think? is it easy? or more complicated?

Reply:
I think it's more difficult to understand than the code itself. After looking at this, I feel like it might be better to make the return doc based on configuration. What do you think about something like this?

```Objective-C

typedef NS_ENUM(uint32_t, TestConflictResolverType) {
    kRemoteWins, kLocalWins, kDeletedWins /*Deleted or Remote*/, kAlwaysDelete /*Return null*/
};

@interface TestConflictResolver: NSObject<CBLConflictResolver>
- (instancetype) initWithType: (TestConflictResolverType)type;
@end

... 

```

Reply:
What about this, which will give more customization by the caller?

```
   TestConflictResolver* resolver;
    resolver = [[TestConflictResolver alloc] initWithResolver: ^CBLDocument* (CBLConflict* con) {
        return con.remoteDocument;
    }];
    pullConfig.conflictResolver = resolver;
```

```
- (instancetype) initWithResolver: (CBLDocument* (^)(CBLConflict*))resolver {
    self = [super init];
    if (self) {
        _resolver = resolver;
    }
    return self;
}
```